doc/modules/bayesian_gaussian_mixture_derivation.rst

```diff
@@ -0,0 +1,628 @@
+scikit this document is not for you.**
+
+This document document presents the equation behind the implementation of
+the :class:`BayesianGaussianMixture` class.
+The API is identical to that of the :class:`GaussianMixture` class.
+
+The inference algorithm is the one from the following book:
+
+    * `Pattern recognition and machine learning
+      <http://www.springer.com/kr/book/9780387310732>`_,
+      Bishop, Christopher M. Springer, Vol. 4 No. 4, 2006.
+
+While the book presents the parts of the variationnl inference algorithm, it
+does not go into detail in the mixture modeling part, which can be just as
+complex, or even more. For this reason we present here a full derivation of the
+inference algorithm and all the update and lower-bound equations.
+
+This implementation is expected to scale at least as well as EM for
+the Gaussian mixture.
+
+Update rules for Variational Bayesian inference
+===============================================
+
+Here the full mathematical derivation of the Variational Bayes update
+rules for Gaussian Mixture Models is given.
+
+Notation
+--------
+During this document, we will use the following statistics :
+
+.. math::
+   :nowrap:
+
+   \begin{eqnarray*}
+      N_k & = & \sum_{n=1}^N {r_{nk}} \\
+      \bar{\mathbf{x}}_k & = & \frac{1}{N_k} \sum_{n=1}^N {r_{nk} \mathbf{x}_n} \\
+      \mathbf{S}_ k & = & \frac{1}{N_k} \sum_{n=1}^N {r_{nk} (\mathbf{x}_n - \bar{\mathbf{x}}_k) (\mathbf{x}_n - \bar{\mathbf{x}}_k)^\top}
+   \end{eqnarray*}
+
+The full model
+--------------
+The model of the Bayesian Gaussian Mixture for full covariance matrices is
+defined by the joint probability :
+
+.. math::
+   :nowrap:
+   :label: vbgmm_joint_probability
+
+   \begin{eqnarray*}
+      p(\mathbf{X}, \mathbf{Z}, \boldsymbol{\pi}, \boldsymbol{\mu}, \boldsymbol{\Lambda}) & = &
+          p(\mathbf{X} | \mathbf{Z}, \boldsymbol{\mu}, \boldsymbol{\Lambda})
+          p(\mathbf{Z} | \boldsymbol{\pi})
+          p(\boldsymbol{\pi})
+          p(\boldsymbol{\mu}|\boldsymbol{\Lambda})
+          p(\boldsymbol{\Lambda}) \\
+   \end{eqnarray*}
+
+where
+
+.. math::
+   :nowrap:
+
+   \begin{eqnarray*}
+      p(\mathbf{X} | \mathbf{Z}, \boldsymbol{\mu}, \boldsymbol{\Lambda})
+        & = & \prod_{n=1}^N \prod_{k=1}^K \mathcal{N} (\mathbf{x}_n | \boldsymbol{\mu}_k, \boldsymbol{\Lambda}_k^{-1})^{z_{nk}} \\
+      p(\mathbf{Z} | \boldsymbol{\pi})
+        & = & \prod_{n=1}^N \prod_{k=1}^K \pi_k^{z_{nk}} \\
+      p(\boldsymbol{\pi})
+        & = & Dirichlet(\boldsymbol{\pi}| \boldsymbol{\alpha}_0) = C(\boldsymbol{\alpha}_0) \prod_{k=1}^K \pi_k^{\alpha_0-1} \\
+      p(\boldsymbol{\mu}, \boldsymbol{\Lambda})
+        & = & p(\boldsymbol{\mu}|\boldsymbol{\Lambda})p(\boldsymbol{\Lambda})  \\
+        & = & \prod_{k=1}^K \mathcal{N}(\boldsymbol{\mu}_k|\mathbf{m}_0, (\beta_0\boldsymbol{\Lambda}_k)^{-1})
+                          \mathcal{W}(\boldsymbol{\Lambda}_k| \mathbf{W}_0, \nu_0)
+   \end{eqnarray*}
```

Comment:
The definition of `z_{nk}` is missing too and also it should probably be `z_{n,k}` right?